lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -865,6 +865,38 @@ def upload_students_csv(_xmodule_instance_args, _entry_id, course_id, task_input
     return task_progress.update_task_state(extra_meta=current_step)
 
 
+def upload_may_enroll_csv(_xmodule_instance_args, _entry_id, course_id, task_input, action_name):
+    """
+    For a given `course_id`, generate a CSV file containing
+    information about students who may enroll but have not done so
+    yet, and store using a `ReportStore`.
+    """
+    start_time = time()
+    start_date = datetime.now(UTC)
+    num_students = CourseEnrollmentAllowed.num_may_enroll_and_unenrolled(course_id)
+    task_progress = TaskProgress(action_name, num_students, start_time)
```

Comment:
I wonder: there's nothing that i can see in the definition of `TaskProgress` that requires the second argument to actually be a number. I wonder if we avoid doing this calculation twice, and instead just insert some string here (eg "all students") or just insert the number 1 (to indicate we're generating 1 report). Frankly this generation will in the majority of cases happen so fast people probably won't even see the Task Progress bar get updated.